nixos/tests/systemd-coredump.nix

```diff
@@ -0,0 +1,16 @@
+import ./make-test-python.nix ({ pkgs, ... }:
+
+{
+  name = "systemd-coredump";
+  meta = with pkgs.lib.maintainers; {
+    maintainers = [ squalus ];
+  };
+
+  nodes.machine = { pkgs, lib, ... }: { };
+
+  testScript = ''
+    machine.wait_for_unit("multi-user.target")
+
+    machine.wait_for_unit("systemd-coredump.socket")
```

Comment:
Can we make this a bit more sophisticated? Try to run a small binary that coredumps, and ensure it's visible when invoking `coredumpctl`?

Reply:
Sure. I'll give that a shot.

Reply:
Improved the test to verify that a crashing service actually shows up in coredumpctl.